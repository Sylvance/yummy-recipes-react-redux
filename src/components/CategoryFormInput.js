import React, { Component } from 'react'
import { Button, Form, FormGroup, Label, Input } from 'reactstrap';
import PropTypes from 'prop-types'

export default class CategoryFormInput extends Component {
  static propTypes = {
    onSave: PropTypes.func.isRequired,
    userid: PropTypes.number.isRequired,
    title: PropTypes.string,
    description: PropTypes.string
  }

  state = {
    userid: this.props.userid || '',
    title: this.props.title || '',
    description: this.props.description || ''
  }

  handleSubmit = e => {
    e.preventDefault();

    const { userid, title, description } = this.state;
    this.props.onSave(userid, title, description)
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value })
  }

  render() {
    return (
      <Form name="form" onSubmit={this.handleSubmit}>
        <FormGroup>
          <Label for="title">Title</Label>
          <Input type="text" 
                name="title" 
                id="title" 
                placeholder="Title" 
                onChange={this.handleChange}
                required />
        </FormGroup>
        <FormGroup>
          <Label for="description">Description</Label>
          <Input type="textarea" 
                name="description" 
                id="description"
                placeholder="Provide a description"
                onChange={this.handleChange}
                required />
        </FormGroup>
        <Button color="success">Add Category</Button>
      </Form>
    )
  }
}
