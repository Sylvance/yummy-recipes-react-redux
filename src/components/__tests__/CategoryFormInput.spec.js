// Auto-generated do not edit

/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from 'react';
import renderer from 'react-test-renderer';
import CategoryFormInput from '../CategoryFormInput';

describe('CategoryFormInput test', () => {
  it('CategoryFormInput should match snapshot', () => {
    const component = renderer.create(
      <CategoryFormInput
        onSave={() => {}}
        userid={42}
        title={'defaultString'}
        description={'defaultString'}
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
