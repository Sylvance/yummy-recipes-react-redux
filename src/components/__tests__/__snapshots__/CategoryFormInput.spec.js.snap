// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryFormInput test CategoryFormInput should match snapshot 1`] = `
<form
  className=""
  name="form"
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="form-control"
      id="title"
      name="title"
      onChange={[Function]}
      placeholder="Title"
      type="text"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className=""
      htmlFor="description"
    >
      Description
    </label>
    <textarea
      className="form-control"
      id="description"
      name="description"
      onChange={[Function]}
      placeholder="Provide a description"
      type="textarea"
    />
  </div>
  <button
    className="btn btn-success"
    onClick={[Function]}
    type={undefined}
  >
    Add Category
  </button>
</form>
`;
